lib/classification_lifecycle.rb

```diff
@@ -14,6 +14,12 @@ def queue(action)
       raise ClassificationNotPersisted.new(message)
     end
     ClassificationWorker.perform_async(classification.id, action)
+    if classification.complete?
+      classification.subject_ids.each do |sid|
+        ClassificationCountWorker.perform_async(sid, classification.workflow_id)
+        RetirementWorker.perform_async(sid, classification.workflow_id)
```

Comment:
Yeah it'd work but it couples the execution of the code to the way we fire off workers, e.g. in the future we run retirement workers as a periodic task and not tied to the incoming classifications (this is a stupid case but it demonstrates the context coupling). Why not just have one worker to call the counter code and retirement together but keep the two classes (remove them as workers and just call them from the single worker)?

Under heavy loads I don't think we can avoid some oversampling based on the way the client loads the subjects and buffers the limit set. 

Also, this will retire the subject but won't remove it from the cellect set for a user, so we would probably overclassify a little anyway under heavy load. We need to call [this](https://github.com/zooniverse/Cellect/blob/4b5341b8fb7dced6e9a8a6cda50cf6c804471595/lib/cellect/client/connection.rb#L23) in the retirement worker to get it out of the active set for users.